src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -1195,81 +1195,91 @@ def run_datamover(self, test_desc=None,
+            pool(TestPool): Pool object.Defaults to None
         """
         # Set the tool to use
         self.set_tool(tool)
 
+
         if create_dataset:
             # create initial datasets
-            pool1 = self.create_pool()
-            cont1 = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+            if not pool:
+                pool = self.create_pool()
+            cont1 = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
 
             # update and run ior on cont1
             self.run_ior_with_params(
                 "DAOS", self.ior_cmd.test_file.value,
-                pool1, cont1)
+                pool, cont1)
+        else:
+            if not pool:
+                pool = self.pool[0]
+            cont1 = self.container[-1]
 
         # create cont2
-        cont2 = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+        cont2 = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
 
         # perform various datamover activities
         if tool == 'CONT_CLONE':
             read_back_cont = self.gen_uuid()
             self.run_datamover(
                 self.test_id + " (cont1 to cont2)",
-                "DAOS", None, pool1, cont1,
-                "DAOS", None, pool1, read_back_cont)
-            read_back_pool = self.pool[0]
+                "DAOS", None, pool, cont1,
+                "DAOS", None, pool, read_back_cont)
+            read_back_pool = pool
         elif tool == 'DSERIAL':
             # Create pool2
-            pool2 = self.create_pool()
+            pool2 = self.get_pool()
             # Use dfuse as a shared intermediate for serialize + deserialize
-            dfuse_cont = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
-            self.start_dfuse(self.dfuse_hosts, pool1, dfuse_cont)
+            dfuse_cont = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
+            self.start_dfuse(self.dfuse_hosts, pool, dfuse_cont)
             self.serial_tmp_dir = self.dfuse.mount_dir.value
 
             # Serialize/Deserialize cont1 to a new cont2 in pool2
             result = self.run_datamover(
                 self.test_id + " (cont1->HDF5->cont2)",
-                "DAOS_UUID", None, pool1, cont1,
+                "DAOS_UUID", None, pool, cont1,
                 "DAOS_UUID", None, pool2, None)
 
             # Get the destination cont2 uuid
             read_back_cont = self.parse_create_cont_uuid(result.stdout_text)
             read_back_pool = pool2
-        else:
+        elif tool in ['FS_COPY', 'DCP']:
             # copy from daos cont1 to cont2
             self.run_datamover(
                 self.test_id + " (cont1 to cont2)",
-                "DAOS", "/", pool1, cont1,
-                "DAOS", "/", pool1, cont2)
+                "DAOS", "/", pool, cont1,
+                "DAOS", "/", pool, cont2)
+        else:
+            self.fail("Invalid tool: {}".format(str(self.tool)))
```

Comment:
I'm noticing the if statement uses `tool` but this check uses `self.tool`. For consistency, we should probably just use `self.tool`, since calling `self.set_tool(tool)` will convert to uppercase.

Reply:
Yeah sure. I have changed it to 'tool' for now. I will change in subsequent PR.